From dbbaa4c43242ba070ff83531c610a5c8b16aad1f Mon Sep 17 00:00:00 2001
From: gejingquan <jingquan.ge@ntu.edu.sg>
Date: Mon, 1 Jul 2024 19:20:33 +0800
Subject: [PATCH] gejingquan update 1

---
 lua.c    |  6 ++++++
 makefile | 10 +++++++---
 2 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/lua.c b/lua.c
index 0f190044..3192e2a5 100644
--- a/lua.c
+++ b/lua.c
@@ -650,8 +650,12 @@ static int pmain (lua_State *L) {
 int main (int argc, char **argv) {
   int status, result;
   lua_State *L = luaL_newstate();  /* create state */
+  printf("Waiting for the hot patch to be installed...\n");
+  sleep(10); 
   if (L == NULL) {
     l_message(argv[0], "cannot create state: not enough memory");
+    printf("Waiting for the hot patch to be uninstalled...\n");
+	sleep(100); 		
     return EXIT_FAILURE;
   }
   lua_pushcfunction(L, &pmain);  /* to call 'pmain' in protected mode */
@@ -661,6 +665,8 @@ int main (int argc, char **argv) {
   result = lua_toboolean(L, -1);  /* get result */
   report(L, status);
   lua_close(L);
+  printf("Waiting for the hot patch to be uninstalled...\n");
+  sleep(100);   
   return (result && status == LUA_OK) ? EXIT_SUCCESS : EXIT_FAILURE;
 }
 
diff --git a/makefile b/makefile
index d46e650c..6554d5a1 100644
--- a/makefile
+++ b/makefile
@@ -66,12 +66,12 @@ LOCAL = $(TESTS) $(CWARNS)
 
 
 # enable Linux goodies
-MYCFLAGS= $(LOCAL) -std=c99 -DLUA_USE_LINUX -DLUA_USE_READLINE
-MYLDFLAGS= $(LOCAL) -Wl,-E
+MYCFLAGS= $(LOCAL) -std=c99 -DLUA_USE_LINUX -DLUA_USE_READLINE -fsanitize=address -fsanitize-recover=address -O0
+MYLDFLAGS= $(LOCAL) -Wl,-E -fsanitize=address
 MYLIBS= -ldl -lreadline
 
 
-CC= gcc
+#CC= gcc
 CFLAGS= -Wall -O2 $(MYCFLAGS) -fno-stack-protector -fno-common -march=native
 AR= ar rc
 RANLIB= ranlib
@@ -121,6 +121,10 @@ clean:
 depend:
 	@$(CC) $(CFLAGS) -MM *.c
 
+install: lua
+	mkdir $$DESTDIR || :
+	cp lua $$DESTDIR
+
 echo:
 	@echo "CC = $(CC)"
 	@echo "CFLAGS = $(CFLAGS)"
-- 
2.34.1

